public/app/plugins/datasource/loki/LanguageProvider.ts

```diff
@@ -1,77 +1,25 @@
-import { chain, difference } from 'lodash';
 import { LRUCache } from 'lru-cache';
-import Prism, { Grammar } from 'prismjs';
+import Prism from 'prismjs';
 
-import { dateTime, AbsoluteTimeRange, LanguageProvider, HistoryItem, AbstractQuery, KeyValue } from '@grafana/data';
-import { CompletionItem, TypeaheadInput, TypeaheadOutput, CompletionItemGroup } from '@grafana/ui';
-import {
-  extractLabelMatchers,
-  parseSelector,
-  processLabels,
-  toPromLikeExpr,
-} from 'app/plugins/datasource/prometheus/language_utils';
+import { LanguageProvider, AbstractQuery, KeyValue } from '@grafana/data';
+import { extractLabelMatchers, processLabels, toPromLikeExpr } from 'app/plugins/datasource/prometheus/language_utils';
 
 import { LokiDatasource } from './datasource';
 import {
   extractLabelKeysFromDataFrame,
   extractLogParserFromDataFrame,
   extractUnwrapLabelKeysFromDataFrame,
 } from './responseUtils';
-import syntax, { FUNCTIONS, PIPE_PARSERS, PIPE_OPERATORS } from './syntax';
+import syntax from './syntax';
 import { ParserAndLabelKeysResult, LokiQuery, LokiQueryType } from './types';
 
-const DEFAULT_KEYS = ['job', 'namespace'];
-const EMPTY_SELECTOR = '{}';
-const HISTORY_ITEM_COUNT = 10;
-const HISTORY_COUNT_CUTOFF = 1000 * 60 * 60 * 24; // 24h
 const NS_IN_MS = 1000000;
 
-// When changing RATE_RANGES, check if Prometheus/PromQL ranges should be changed too
-// @see public/app/plugins/datasource/prometheus/promql.ts
-const RATE_RANGES: CompletionItem[] = [
-  { label: '$__auto', sortValue: '$__auto' },
-  { label: '1m', sortValue: '00:01:00' },
-  { label: '5m', sortValue: '00:05:00' },
-  { label: '10m', sortValue: '00:10:00' },
-  { label: '30m', sortValue: '00:30:00' },
-  { label: '1h', sortValue: '01:00:00' },
-  { label: '1d', sortValue: '24:00:00' },
-];
-
-export const LABEL_REFRESH_INTERVAL = 1000 * 30; // 30sec
-
-const wrapLabel = (label: string) => ({ label, filterText: `\"${label}\"` });
-
-export type LokiHistoryItem = HistoryItem<LokiQuery>;
-
-type TypeaheadContext = {
-  history?: LokiHistoryItem[];
-  absoluteRange?: AbsoluteTimeRange;
-};
-
-export function addHistoryMetadata(item: CompletionItem, history: LokiHistoryItem[]): CompletionItem {
-  const cutoffTs = Date.now() - HISTORY_COUNT_CUTOFF;
-  const historyForItem = history.filter((h) => h.ts > cutoffTs && h.query.expr === item.label);
-  let hint = `Queried ${historyForItem.length} times in the last 24h.`;
-  const recent = historyForItem[0];
-
-  if (recent) {
-    const lastQueried = dateTime(recent.ts).fromNow();
-    hint = `${hint} Last queried ${lastQueried}.`;
-  }
-
-  return {
-    ...item,
-    documentation: hint,
-  };
-}
-
```

Comment:
All of this code was for slate editor to suggest previously run queries in autocomplete, but we don't use it anymore. All of the autocompletion logic is now in `CompletionDataProvider`, so it is safe to remove. https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/components/monaco-query-field/monaco-completion-provider/CompletionDataProvider.ts#L32